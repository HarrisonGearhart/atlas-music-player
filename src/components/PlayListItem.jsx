import React from "react";

export default function PlayListItem({ title, artist, length, selected }) {
  return (
    <div
      className={`flex justify-between items-center px-4 py-2 rounded-md cursor-pointer transition-colors duration-200 ${
        selected
          ? "bg-emerald dark:bg-amethyst"
          : "hover:bg-amethyst dark:hover:bg-emerald"
      }`}
    >
      <div>
        <p className="text-sm font-medium">{title}</p>
        <p className="text-xs">{artist}</p>
      </div>
      <span className="text-xs">{length}</span>
    </div>
  );
}
