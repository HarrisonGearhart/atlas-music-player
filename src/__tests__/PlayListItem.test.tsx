import React from "react";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import PlayListItem from "../components/PlayListItem";

describe("PlayListItem Component", () => {
  it("renders with default props", () => {
    const { container } = render(<PlayListItem />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders selected item with custom highlight", () => {
    const { container } = render(
      <PlayListItem selected title="Test" artist="Artist" length="4:00" highlightColor="#2ecc71" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders unselected item with custom props", () => {
    const { container } = render(
      <PlayListItem title="Song" artist="Someone" length="3:30" selected={false} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
