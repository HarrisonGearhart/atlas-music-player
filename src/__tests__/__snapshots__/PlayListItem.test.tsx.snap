// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem Component > renders selected item with custom highlight 1`] = `
<div
  class="flex justify-between items-center px-4 py-2 cursor-pointer transition-colors"
  style="background-color: rgb(46, 204, 113); color: rgb(255, 255, 255);"
>
  <div
    class="flex flex-col"
  >
    <p
      class="text-sm font-medium"
    >
      Test
    </p>
    <p
      class="text-xs"
    >
      Artist
    </p>
  </div>
  <span
    class="text-xs"
  >
    4:00
  </span>
</div>
`;

exports[`PlayListItem Component > renders unselected item with custom props 1`] = `
<div
  class="flex justify-between items-center px-4 py-2 cursor-pointer transition-colors"
>
  <div
    class="flex flex-col"
  >
    <p
      class="text-sm font-medium"
    >
      Song
    </p>
    <p
      class="text-xs"
    >
      Someone
    </p>
  </div>
  <span
    class="text-xs"
  >
    3:30
  </span>
</div>
`;

exports[`PlayListItem Component > renders with default props 1`] = `
<div
  class="flex justify-between items-center px-4 py-2 cursor-pointer transition-colors"
>
  <div
    class="flex flex-col"
  >
    <p
      class="text-sm font-medium"
    >
      Song Title
    </p>
    <p
      class="text-xs"
    >
      Artist
    </p>
  </div>
  <span
    class="text-xs"
  >
    3:45
  </span>
</div>
`;
